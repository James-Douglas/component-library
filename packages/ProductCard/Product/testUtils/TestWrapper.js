import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
