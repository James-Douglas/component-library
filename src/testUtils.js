// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
