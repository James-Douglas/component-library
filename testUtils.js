import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
