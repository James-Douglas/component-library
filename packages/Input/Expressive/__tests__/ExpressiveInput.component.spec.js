import React from 'react';
import 'jest-styled-components';
import { useTracking } from 'react-tracking';
import { render, fireEvent, wait } from '../../../../testUtils';
import ExpressiveInput from '../ExpressiveInput.component';

describe('ExpressiveInput', () => {
  it('renders with minimal props', () => {
    const { container } = render(<ExpressiveInput id="test-id" handleChange={() => {}} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders with lots of props', () => {
    const { container, getByText } = render(
      <ExpressiveInput
        trackingLabel="test"
        id="test-id"
        label="label-test"
        placeholder="placeholder test"
        prefixContent="$"
        maxlength={5}
        autocomplete="on"
        handleChange={() => {}}
      />,
    );
    const expressiveInput = container.querySelector('input');
    const id = expressiveInput.getAttribute('id');
    const label = container.querySelector('label');
    const placeholder = expressiveInput.getAttribute('placeholder');
    const autocomplete = expressiveInput.getAttribute('autocomplete');

    expect(id).toBe('test-id');
    expect(label.textContent).toBe('label-test');
    expect(placeholder).toBe('placeholder test');
    expect(getByText('$')).toBeInTheDocument();
    expect(autocomplete).toBe('on');
  });

  it('renders the expected design', () => {
    const { container, getByText } = render(
      <ExpressiveInput
        trackingLabel="test"
        id="test-id"
        label="label-test"
        handleChange={() => {}}
      />,
    );

    const inputField = container.querySelector('input');
    const label = getByText('label-test');

    expect(label).toHaveStyleRule('visibility', 'hidden');

    fireEvent.input(inputField, { target: { value: 'test' } });

    wait(() => {
      expect(label).toHaveStyleRule('visibility', 'visible');
    });
  });

  it('calls func on change', () => {
    const handlerFn = jest.fn();
    const { container } = render(
      <ExpressiveInput
        trackingLabel="test"
        id="test-id"
        handleChange={handlerFn}
      />,
    );

    const inputField = container.querySelector('input');
    fireEvent.input(inputField, { target: { value: 'test' } });

    expect(handlerFn).toHaveBeenCalled();
    expect(handlerFn.mock.calls.length).toBe(1);
    expect(handlerFn.mock.calls[0][0]).toBe('test');
  });

  describe('interaction tracking', () => {
    it('tracks focus events', () => {
      const { trackEvent } = useTracking();
      const { container } = render(
        <ExpressiveInput
          trackingLabel="test"
          id="test-id"
          label="label-test"
          placeholder="placeholder test"
          handleChange={() => {}}
        />,
      );
      const inputField = container.querySelector('#test-id');
      fireEvent.focus(inputField);
      expect(trackEvent).toHaveBeenCalledWith({
        interaction: {
          ixn_action: 'Focus',
          ixn_label: 'test',
          ixn_object: 'Input',
          ixn_type: 'Expressive',
          ixn_value: '',
        },
      });
    });

    it('tracks input events', () => {
      const { trackEvent } = useTracking();
      const { container } = render(
        <ExpressiveInput
          trackingLabel="test"
          id="test-id"
          label="label-test"
          placeholder="placeholder test"
          handleChange={() => {}}
        />,
      );
      const inputField = container.querySelector('#test-id');
      fireEvent.input(inputField, { target: { value: 'test content' } });
      expect(trackEvent).toHaveBeenCalledWith({
        interaction: {
          ixn_action: 'Input',
          ixn_label: 'test',
          ixn_object: 'Input',
          ixn_type: 'Expressive',
          ixn_value: 'test content',
        },
      });
    });
  });
});
