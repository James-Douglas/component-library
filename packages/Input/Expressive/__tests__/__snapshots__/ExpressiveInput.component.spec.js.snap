// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpressiveInput renders with minimal props 1`] = `"<div class=\\"sc-fzpans mKmoT input-wrap\\"><div class=\\"sc-fznKkj bEKlFj input-container\\"><div class=\\"sc-fznZeY gCjgZb\\"><div class=\\"sc-fzokOt cXBDVM input-clear-wrap\\"><div class=\\"sc-AxgMl hTWZYh\\"><label id=\\"\\" for=\\"test-id\\" class=\\"sc-AxheI iCGNxP label\\"></label></div><input id=\\"test-id\\" name=\\"test-id\\" type=\\"text\\" placeholder=\\"\\" aria-labelledby=\\"test-id-label\\" aria-describedby=\\"test-id-tooltip\\" autocomplete=\\"off\\" class=\\"sc-fzqBZW cjwyNe input-default \\" breakpoint=\\"md\\" value=\\"\\"></div></div></div></div>"`;
